packages/flutter/lib/src/widgets/single_child_scroll_view.dart

```diff
@@ -396,6 +396,10 @@ class _RenderSingleChildViewport extends RenderBox with RenderObjectWithChildMix
   }
 
   void _hasScrolled() {
+    // Content dimensions do not change as SingleChildScrollView only lays out
+    // once, but the metrics can change based on scrolling and we may want to
+    // dispatch a ScrollMetricsNotification.
+    offset.applyContentDimensions(_minScrollExtent, _maxScrollExtent);
```

Comment:
According to the original design intent, `ScrollMetricsNotification` is to complement `ScrollNotification`, not to replace `ScrollNotification`. As the documentation says
```
/// The above behaviors usually do not trigger [ScrollNotification] events,
/// so this is useful for listening to [ScrollMetrics] changes that are not
/// caused by the user scrolling.
```
But would love to explore further how to revamp it.  : ）